tests/js/client/communication/test-communication.js

```diff
@@ -470,95 +488,68 @@ function GenericAqlSetupPathSuite(type) {
     });
   `;
 
-  const apiLibs = `
-    const request = require("@arangodb/request");
-    arango.getEndpoint()
-    function sendRequest(method, endpoint, body, headers) {
-
-      const endpointToURL = (endpoint) => {
-        if (endpoint.substr(0, 6) === 'ssl://') {
-          return 'https://' + endpoint.substr(6);
-        }
-        var pos = endpoint.indexOf('://');
-        if (pos === -1) {
-          return 'http://' + endpoint;
-        }
-        return 'http' + endpoint.substr(pos);
-      };
-
-      let res;
-      try {
-        const envelope = {
-          json: true,
-          method,
-          url: endpointToURL(arango.getEndpoint()) + endpoint,
-          headers,
-        };
-        if (method !== 'GET') {
-          envelope.body = body;
-        }
-        res = request(envelope);
-      } catch (err) {
-        require("console").log(err);
-        return {};
-      }
-      if (typeof res.body === "string") {
-        if (res.body === "") {
-          res.body = {};
-        } else {
-          res.body = JSON.parse(res.body);
-        }
-      }
-      return res;
-    }
-  `;
   // Note: A different test checks that the API works this way
   const apiExclusive = `
-    ${apiLibs}
     let trx;
     const obj = { collections: { exclusive: "${twoShardColName}" } };
-    let result = sendRequest('POST', "/_api/transaction/begin", obj, {}, true);
-    try {
-      trx = result.body.result.id;
+    let result = arango.POST_RAW("/_api/transaction/begin", obj);
+    if (result.code === 201) {
+      trx = result.parsedBody.result.id;
       const query = "FOR x IN 1..${docsPerWrite} INSERT {} INTO ${twoShardColName}";
-      sendRequest('POST', "/_api/cursor", { query }, { "x-arango-trx-id": trx });
-      // Commit
-      sendRequest('PUT', '/_api/transaction/' + encodeURIComponent(trx), {}, {}, true);
-    } catch {
-      // Abort
-      sendRequest('DELETE', '/_api/transaction/' + encodeURIComponent(trx), {}, {}, true);
+      result = arango.POST_RAW("/_api/cursor", { query }, { "x-arango-trx-id": trx });
+      if (result.code === 201) {
+        // Commit
+        result = arango.PUT_RAW('/_api/transaction/' + encodeURIComponent(trx), {}, {});
+      }
+      if (result.code !== 200) {
+        print('apiExclusive failure:');
+        print(result);
+        print(arango.DELETE_RAW('/_api/transaction/' + encodeURIComponent(trx), {}, {}));
+      }
+    } else {
+      print(result.parsedBody);
```

Comment:
Is there a reason here (and in the following 2 cases) to only print errors/responses instead of making the test fail?